discovery/file/file_test.go

```diff
@@ -29,3 +29,16 @@ func TestRegister(t *testing.T) {
 	discovery := &FileDiscoveryService{path: "/path/to/file"}
 	assert.Error(t, discovery.Register("0.0.0.0"))
 }
+
+func TestContentWithComment(t *testing.T) {
+	data := `
+### test ###
+1.1.1.1:1111 # inline comment
+# 2.2.2.2:2222
+      ### empty line with comment
+### test ###
+`
+	ips := parseFileContent([]byte(data))
+	assert.Equal(t, 1, len(ips))
+	assert.Equal(t, ips[0], "1.1.1.1:1111")
```

Comment:
To conform with the above, this line should write:
`assert.Equal(t, "1.1.1.1:1111", ips[0])`
because the expected value came first.